python/lsst/ts/externalscripts/maintel/correct_pointing.py

```diff
@@ -0,0 +1,436 @@
+        target_ra_deg = Angle(target_ra_rad, unit=u.rad).to(u.deg).value
+        target_dec_deg = Angle(target_dec_rad, unit=u.rad).to(u.deg).value
+
+        self.log.info(
+            f"Target coordinates: RA={target_ra_deg:.6f} deg, Dec={target_dec_deg:.6f} deg"
+        )
+
+        offset_magnitude_arcsec = float("inf")
+
+        for iteration in range(1, self.max_iterations + 1):
+            if offset_magnitude_arcsec < self.tolerance_arcsec:
+                self.log.info(
+                    f"Pointing offset {offset_magnitude_arcsec:.3f} arcsec is within "
+                    f"tolerance {self.tolerance_arcsec:.3f} arcsec. Correction complete."
+                )
+                break
+            else:
+                self.log.info(
+                    f"Starting iteration {iteration} of {self.max_iterations}"
+                )
+
+            exposure_ids = await self.lsstcam.take_acq(
+                exptime=self.exposure_time,
+                n=1,
+                group_id=self.group_id,
+                reason="pointing_correction",
+                filter=self.filter,
+            )
+
+            exposure_id = int(exposure_ids[0])
+            self.log.info(f"Took exposure {exposure_id}")
+
+            measured_ra_deg, measured_dec_deg = (
+                await self.get_measured_coordinates_from_consdb(exposure_id)
+            )
+
+            offset_ra_arcsec, offset_dec_arcsec, offset_magnitude_arcsec = (
+                self.calculate_offset(
+                    target_ra_deg, target_dec_deg, measured_ra_deg, measured_dec_deg
+                )
+            )
+
+            self.log.info(
+                f"Measured coordinates: RA={measured_ra_deg:.6f} deg, "
+                f"Dec={measured_dec_deg:.6f} deg\n"
+                f"Offset: RA={offset_ra_arcsec:.3f} arcsec, "
+                f"Dec={offset_dec_arcsec:.3f} arcsec, "
+                f"Magnitude={offset_magnitude_arcsec:.3f} arcsec"
+            )
+
+            if offset_magnitude_arcsec >= self.tolerance_arcsec:
+                await self.apply_pointing_offset(offset_ra_arcsec, offset_dec_arcsec)
+            else:
+                self.log.info(
+                    f"Pointing offset {offset_magnitude_arcsec:.3f} arcsec is within "
+                    f"tolerance {self.tolerance_arcsec:.3f} arcsec. Correction complete."
+                )
+        else:
+            raise RuntimeError(
+                f"Failed to correct pointing after {self.max_iterations} iterations. "
+                f"Final offset: {offset_magnitude_arcsec:.3f} arcsec"
+            )
+
+        self.log.info("Pointing correction successful.")
+
+    async def get_target_coordinates(self):
+        """Get the current target coordinates from MTPtg.
+
+        Returns
+        -------
+        tuple
+            Target RA and Dec in radians (ra_rad, dec_rad).
+        """
+        try:
+            current_target = await self.mtcs.rem.mtptg.evt_currentTarget.aget(
```

Comment:
You need to add `currentTarget` to the `Slew` usages in MTCS [here](https://github.com/lsst-ts/ts_observatory_control/blob/53110832abea07264457663e5a92e6ac56f84388/python/lsst/ts/observatory/control/maintel/mtcs.py#L3773).